src/URDFConverter/UrdfUnity/UrdfUnity/FileManager/FileManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.IO;
+using UrdfUnity.Urdf.Models;
+using UrdfUnity.Util;
+using UrdfUnity.Parse;
+using NLog;
+
+
+namespace UrdfUnity.FileManager
+{
+    /// <summary>
+    /// Filetypes we care about in the file manager.
+    /// </summary>
+    public enum FileType
+    {
+        UNKNOWN,
+        URDF,
+        XACRO
+    }
+
+
+    /// <summary>
+    /// This class will be handling the FileIO 
+    /// </summary>
+    public class FileManager
+    {
+        Logger LOGGER;
+        
+
+        /// <summary>
+        /// Base constructor, for initializing components
+        /// </summary>
+        public FileManager()
+        {
+            // initializing the logger for this class.
+            LOGGER = LogManager.GetCurrentClassLogger();
+        }
+
+
+        /// <summary>
+        /// We have a URDF/XACRO filepath that needs to be extracted to be
+        /// a Robot model. 
+        /// </summary>
+        /// <param name="filePath">Path of file containing urdf/xacro for the robot</param>
+        /// <returns>
+        ///     Robot|null - A Robot model if a valid file path and content for a robot. 
```

Comment:
Nitpick++; Following lead of other <returns>: "The Robot model parsed from the specified filepath, otherwise <c>null<c>

Unless you borrowed this cue from MSDN documentation, in which case point me to it!